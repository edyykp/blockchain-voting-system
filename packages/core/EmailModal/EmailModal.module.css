.container {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 50px;
  flex-direction: column;
  background-color: var(--gray-700);
  height: 300px;
}
