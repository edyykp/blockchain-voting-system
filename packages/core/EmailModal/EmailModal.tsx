import { useRef, useState } from 'react';
import { useRouter } from 'next/router';

import { Button, Input, Modal } from '@packages/components';
import { registerUser } from '@packages/network';

import styles from './EmailModal.module.css';
import { useEmailModalContext } from '../../config/SetEmailModalContext/state';

export type EmailModalProps = {
  show: boolean;
  setShowModal: (show: boolean) => void;
  walletAddress?: string;
};

export const EmailModal = ({
  show,
  setShowModal,
  walletAddress,
}: EmailModalProps) => {
  const emailRef = useRef<HTMLInputElement>(null);
  const [error, setError] = useState<string | undefined>();
  const router = useRouter();
  const { setShow } = useEmailModalContext();

  const storeEmail = async () => {
    if (!emailRef.current?.value) {
      setError('Please insert an email address');
    }
    const { error, status } = await registerUser(
      emailRef.current?.value!,
      '',
      walletAddress || null,
    );

    if (status === 201) {
      setShow(false);
      router.push('/dashboard');
      return;
    }

    if (status !== 201) {
      setError(error);
    }
  };

  return (
    <Modal
      show={show}
      onClose={() => setShowModal(false)}
      title="Set up your email"
      children={
        <div className={styles.container} data-testid="email-modal">
          {error && <div className={styles.errorWrapper}>{error}</div>}
          <Input field="email" type="text" isRequired inputRef={emailRef} />
          <Button
            theme="primary"
            size="lg"
            buttonText="Set email and log in"
            onClick={(event) => {
              event?.preventDefault();
              storeEmail();
            }}
          />
        </div>
      }
      theme="dark"
    />
  );
};
