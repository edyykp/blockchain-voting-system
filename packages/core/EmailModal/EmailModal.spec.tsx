import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { EmailModal, EmailModalProps } from './EmailModal';

const MOCK_PROPS_NO_SHOW = {
  show: false,
  driverName: 'Charles Leclerc',
  setShowModal: () => {},
  voteFinishedCallback: () => {},
};

const MOCK_PROPS_SHOW = {
  show: true,
  driverName: 'Charles Leclerc',
  setShowModal: () => {},
  voteFinishedCallback: () => {},
};

const setup = (props: EmailModalProps) => {
  return {
    ...render(<EmailModal {...props} />),
  };
};

describe('EmailModalâ†’', () => {
  it('renders', async () => {
    const { container, getByTestId } = setup(MOCK_PROPS_SHOW);

    expect(getByTestId('email-modal')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('renders nothing if show is false', async () => {
    const { container, queryByTestId } = setup(MOCK_PROPS_NO_SHOW);

    expect(queryByTestId('email-modal')).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
