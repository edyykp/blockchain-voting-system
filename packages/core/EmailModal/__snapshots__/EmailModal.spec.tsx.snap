// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailModal→ renders 1`] = `
<div>
  <div
    class="overlay"
  >
    <div
      class="wrapper darkBackground"
    >
      <div
        class="header"
      >
        <h1
          class="title"
        >
          Set up your email
        </h1>
        <svg
          class="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
      </div>
      <div
        class="body"
      >
        <div
          class="container"
          data-testid="email-modal"
        >
          <div
            class="container"
          >
            <input
              autocomplete="off"
              class="input"
              placeholder="Enter your email"
              required=""
              type="text"
            />
            <label
              class="label "
              for="email"
            >
              Email
            </label>
          </div>
          <button
            class="button  primary
      false
      false
      largeButton
      "
          >
            <span>
              Set email and log in
            </span>
             
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmailModal→ renders nothing if show is false 1`] = `<div />`;
